src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenOptimizedNullableOperators.cs

```diff
@@ -2414,5 +2414,371 @@ .maxstack  2
+                  IL_000d:  ret
+                }
+                """);
+        }
+
+        [Fact, WorkItem("https://github.com/dotnet/roslyn/issues/66431")]
+        public void NullableConstant_Int_E0()
+        {
+            NullableConstant_Int("x == 0", "1000", """
+                {
+                  // Code size       23 (0x17)
+                  .maxstack  2
+                  .locals init (int? V_0,
+                                int V_1)
+                  IL_0000:  ldarg.0
+                  IL_0001:  stloc.0
+                  IL_0002:  ldc.i4.0
+                  IL_0003:  stloc.1
+                  IL_0004:  ldloca.s   V_0
+                  IL_0006:  call       "int int?.GetValueOrDefault()"
+                  IL_000b:  ldloc.1
+                  IL_000c:  ceq
+                  IL_000e:  ldloca.s   V_0
+                  IL_0010:  call       "bool int?.HasValue.get"
+                  IL_0015:  and
+                  IL_0016:  ret
+                }
+                """);
+        }
+
+        [Fact, WorkItem("https://github.com/dotnet/roslyn/issues/66431")]
+        public void NullableConstant_Int_0E()
+        {
+            NullableConstant_Int("0 == x", "1000", """
+                {
+                  // Code size       21 (0x15)
+                  .maxstack  2
+                  .locals init (int? V_0)
+                  IL_0000:  ldc.i4.0
+                  IL_0001:  ldarg.0
+                  IL_0002:  stloc.0
+                  IL_0003:  ldloca.s   V_0
+                  IL_0005:  call       "int int?.GetValueOrDefault()"
+                  IL_000a:  ceq
+                  IL_000c:  ldloca.s   V_0
+                  IL_000e:  call       "bool int?.HasValue.get"
+                  IL_0013:  and
+                  IL_0014:  ret
+                }
+                """);
+        }
+
+        [Fact, WorkItem("https://github.com/dotnet/roslyn/issues/66431")]
+        public void NullableConstant_Int_L1()
+        {
+            NullableConstant_Int("x < 1", "1000", """
+                {
+                  // Code size       23 (0x17)
+                  .maxstack  2
+                  .locals init (int? V_0,
+                                int V_1)
+                  IL_0000:  ldarg.0
+                  IL_0001:  stloc.0
+                  IL_0002:  ldc.i4.1
+                  IL_0003:  stloc.1
+                  IL_0004:  ldloca.s   V_0
+                  IL_0006:  call       "int int?.GetValueOrDefault()"
+                  IL_000b:  ldloc.1
+                  IL_000c:  clt
+                  IL_000e:  ldloca.s   V_0
+                  IL_0010:  call       "bool int?.HasValue.get"
+                  IL_0015:  and
+                  IL_0016:  ret
+                }
+                """);
+        }
+
+        [Fact, WorkItem("https://github.com/dotnet/roslyn/issues/66431")]
+        public void NullableConstant_Int_GE1()
+        {
+            NullableConstant_Int("x >= -1", "1110", """
```

Comment:
>NullableConstant_Int [](http://example.com/codeflow?start=12&length=20)

I am not going to block on this, but, I think the tests  would be easier to read and maintain if we were not using helpers like that, but instead had each unit-test explicitly test its own scenario as other tests in this file do. I asked to split up a single monolithic unit test into a series of unit-tests, each covering one specific scenario. The motivation for suggestion was better readability and ease of maintenance. I don't feel like I got what I was asking for. I feel like we went from one obscure way of testing things to another, perhaps, even more obscure way of testing things.